build.gradle

```diff
@@ -19,40 +19,78 @@ ext {
 }
 
-subprojects {
-    apply plugin: 'maven-publish'  // Plugin to publish artifacts to a maven repo
-    apply plugin: 'signing'
-    apply plugin: 'java'
+// Why is this necessary? Who knows!
+tasks.register('clean', Delete) {
+    delete rootProject.buildDir
+}
 
-    task cloneAndPatchGraal(type: Exec) {
+task cloneGraal(type: Exec) {
 
-        // Task details
-        group = 'Other'
-        description = 'Clones a specific version of Graal and patches the source code with the required changes for Conclave.'
+    // Task details
+    group = 'Other'
+    description = 'Clone a specific version of Graal. Delete existing graal clone if present.'
 
-        def patchingScript = "${rootProject.projectDir}/scripts/cloneAndPatch.sh"
-        def graalPatch = "${rootProject.projectDir}/patches/graal.patch"
+    def cloneScript = "${rootProject.projectDir}/scripts/cloneGraal.sh"
 
-        inputs.files(patchingScript, graalPatch)
-        outputs.files(graalCloneDummy)
+    // Re-run this only if the version or commit ID have changed.
+    inputs.property('graalVersion', graal_version)
+    inputs.property('graalCommitID', graal_commit_id)
+    inputs.file(cloneScript)
 
-        doFirst {
-            mkdir tempDir
-            workingDir tempDir
-            commandLine "$patchingScript", "$graal_version", "$graal_commit_id"
-            //commandLine "ls"
+    // Used to detect if subsequent tasks should re-run
+    outputs.file(graalCloneDummy)
 
-            // Write current date to graal clone dummy file. If this task is run again, the content of this file
-            // will change and subsequent tasks will run.
-            writeDateToFile(graalCloneDummy)
-        }
+    doFirst {
+        mkdir tempDir
+        workingDir tempDir
+        commandLine "$cloneScript", "$graal_version", "$graal_commit_id"
+    }
+
+    doLast {
+        // Write current date-time to dummy file. If this task is run again, the content of this file
+        // will change and subsequent tasks will re-run.
+        writeDateToFile(graalCloneDummy)
+    }
+}
+
+task patchGraal(type: Exec) {
+    dependsOn(cloneGraal)
+
+    // Task details
+    group = 'Other'
+    description = 'Patch the graal repository with Conclave specific changes.'
+
+    def patchScript = "${rootProject.projectDir}/scripts/patchGraal.sh"
+    def patchFile = "${rootProject.projectDir}/patches/graal.patch"
+
+    inputs.files(patchScript, patchFile, graalCloneDummy)
+    outputs.file(graalPatchDummy)
+
+    doFirst {
+        mkdir tempDir
+        workingDir tempDir
+        commandLine "$patchScript"
     }
 
+    doLast {
+        // Write current date-time to dummy file. If this task is run again, the content of this file
+        // will change and subsequent tasks will re-run.
+        writeDateToFile(graalPatchDummy)
+    }
+}
+
+subprojects {
+    apply plugin: 'maven-publish'  // Plugin to publish artifacts to a maven repo
+    apply plugin: 'signing'
+    apply plugin: 'java'
```

Comment:
As you can see at line 89, these plugins are only applied to subprojects also known as modules. When the command `clean` is run at the root level, Gradle will look and run all the `clean` tasks from the subprojects even if the root project does not have a `clean` task. Gradle will only throw an error if the root project and subprojects don't have the `clean` task.